app/lib/features/chat/widgets/custom_input.dart

```diff
@@ -607,15 +609,31 @@ class _FileWidget extends ConsumerWidget {
 }
 
 class _TextInputWidget extends ConsumerWidget {
+  final GlobalKey<FlutterMentionsState> mentionKey;
   final Convo convo;
   final Function() onSendButtonPressed;
   final bool isEncrypted;
   const _TextInputWidget({
+    required this.mentionKey,
     required this.convo,
     required this.onSendButtonPressed,
     this.isEncrypted = false,
   });
 
+  void _updateTextValue(String roomId, WidgetRef ref) {
+    var textValues = ref.watch(_textValuesProvider);
+
+    if (textValues[roomId] != null) {
+      textValues[roomId] = '';
+      String value = textValues[roomId]!;
+      value += mentionKey.currentState!.controller!.text;
+      textValues[roomId] = value;
+    } else {
+      textValues[roomId] = mentionKey.currentState!.controller!.text;
+    }
+    ref.read(_textValuesProvider.notifier).update((state) => textValues);
```

Comment:
Using a family would also shorten this to:

```suggestion
    var textValues = ref.watch(_textValuesProvider);
    textValues += mentionKey.currentState!.controller!.text;
    ref.read(_textValuesProvider(roomId).notifier).update((state) => textValues);
```